src/Microsoft.Data.Analysis/TextFieldParser.cs

```diff
@@ -0,0 +1,642 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.Data.Analysis
+{
+    public enum FieldType
+    {
+        Delimited,
+        FixedWidth
+    }
+
+    public class TextFieldParser : IDisposable
```

Comment:
Does this need to be public?

Reply:
Not necessarily. 
1) I wasn't sure if we wanted to make `TextFieldParser` it's own Nuget package. 
2) It was easier to test with it being public. 

Made it internal now. Also using internalsvisibleto for the tests